docs/docs/layout-engines.md

```diff
@@ -0,0 +1,89 @@
+# Layout Engines
```

Comment:
I have mixed feelings about `ColumnLayoutEngine` - I feel the naming is inconsistent with `SliceLayoutEngine`. I called it `ColumnLayoutEngine` because it lays out windows in a row of windows, which looked like columns. I now think this is the wrong name...

I was inclined to create a new one correctly named, but I think I like your idea of adding new configs better.